### PR TITLE
add ignorePodsWithoutPDB option

### PR DESCRIPTION
**Pull request by john7doe**
This PR adds a new option to the default evictor plugin called `ignorePodsWithoutPDB` when set to `true` it will filter out pods without PDBs.

The option was added to make it easier to ensure that the descheduler only targets pods that has explicitly defined their disruption budget.

### CODE REVIEW

**Comment by john7doe** [comment]
@ingvagabund 
New to the codebase, so did not think of running the e2e tests before committing. I ran manual tests from the cli with my credentials, which explains why there were no issues fetching the Pdb's. 

Will fix this and ping you once the tests are green.

**Comment by john7doe** [comment]
@ingvagabund e2e test passes locally now :-)

**Comment by john7doe** [comment]
@ingvagabund I think someone needs to update the required checks (remove  pull-descheduler-test-e2e-k8s-master-1-28)

**Review by ingvagabund — COMMENTED** [review]
Few nits. Otherwise looks great in overall. This is quite a nice addition to the codebase. Thank you for your contribution.

File: pkg/framework/plugins/defaultevictor/defaultevictor.go [hunk 1]
```diff
@@ -195,6 +195,17 @@ func New(args runtime.Object, handle frameworktypes.Handle) (frameworktypes.Plug
 			return nil
 		})
 	}
+
+	if defaultEvictorArgs.IgnorePodsWithoutPDB {
+		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
+			hasPdb, err := utils.IsPodCoveredByPDB(pod, handle.SharedInformerFactory().Policy().V1().PodDisruptionBudgets().Lister())
+			if err != nil || !hasPdb {
+				return fmt.Errorf("no PodDisruptionBudget found for pod")
```
[the remark this answers is not in the document]
Comment (john7doe) [reply pkg/framework/plugins/defaultevictor/defaultevictor.go:203]: I can do that, in fact I did that in the initial pass on the code. The `PodDisruptionBudgetLister` implementation has a, in my opinion :-) design choice that less that ideal: https://github.com/john7doe/descheduler/blob/master/vendor/k8s.io/client-go/listers/policy/v1/poddisruptionbudget_expansion.go#L63-L65

TL;DR no pdb is returned as an error :-(

So logging errors would be quite verbose. Its also a plain error, so `errors.Is` is not an option to differentiate between actual errors and `len() == 0` ones. 

One option could be to log at a very high verbosity level, so that the errors are normally no logged, but can be seen if someone needs to investigate.

[the remark this answers is not in the document]
Comment (ingvagabund) [reply pkg/framework/plugins/defaultevictor/defaultevictor.go:203]: I see. That's unfortunate. In that case it will be better to call `PodDisruptionBudgets.List` directly/copy paste some of the code from https://github.com/john7doe/descheduler/blob/master/vendor/k8s.io/client-go/listers/policy/v1/poddisruptionbudget_expansion.go#L63-L65.

[the remark this answers is not in the document]
Comment (john7doe) [reply pkg/framework/plugins/defaultevictor/defaultevictor.go:206]: @ingvagabund done, please have a look

**Comment by ingvagabund** [comment]
Thank you for addressing all the review comments.
/approve
/lgtm

**Comment by john7doe** [comment]
Thanks for accepting :-) I think this still needs to happen: https://github.com/kubernetes-sigs/descheduler/pull/1529#issuecomment-2404486529

Br,
/Simon
